### PR TITLE
[VirtualCategories] Option to force empty category if virtual category root disabled

### PR DESCRIPTION
**Pull request by rbayet**
Work in Progress

### CODE REVIEW

File: src/module-elasticsuite-virtual-category/Model/Rule.php [hunk 1]
```diff
@@ -189,6 +198,7 @@ public function getCategorySearchQuery($category, $excludedCategories = []): ?Qu
             ]
         );
 
+        $query = false;
```
Comment (rbayet) [inline src/module-elasticsuite-virtual-category/Model/Rule.php:201]: Why is this necessary ?

File: src/module-elasticsuite-virtual-category/Model/Rule.php [hunk 2]
```diff
@@ -306,9 +316,14 @@ private function buildCategorySearchQuery($category, $excludedCategories = []):
         if (!in_array($category->getId(), $excludedCategories)) {
             $excludedCategories[] = $category->getId();
 
-            if ((bool) $category->getIsVirtualCategory() && $category->getIsActive()) {
+            if (!$category->getIsActive()
+                && $this->config->isForceZeroResultsForDisabledCategoriesEnabled($this->getStoreId())) {
+                return $this->getNoResultsQuery();
+            }
+
+            if ($category->getIsVirtualCategory()) {
```
Comment (rbayet) [inline src/module-elasticsuite-virtual-category/Model/Rule.php:324]: Be careful, you are creating a specific route that is not correctly covered.
You cannot remove the test on `$category->getIsActive()` on lines 324 and 326 since the test on 319 also relies on the new setting.
So if "isForceZero..." returns false, the category could be disabled on lines 324 and 326...

File: src/module-elasticsuite-virtual-category/Model/Rule.php [hunk 3]
```diff
@@ -189,6 +198,7 @@ public function getCategorySearchQuery($category, $excludedCategories = []): ?Qu
             ]
         );
```
Comment (rbayet) [inline src/module-elasticsuite-virtual-category/Model/Rule.php:199]: Shouldn't the value of the setting be taken into account for the cacheKey ?

File: src/module-elasticsuite-virtual-category/Model/Rule.php (hunk 2 above)
Comment (PierreGauthier) [reply src/module-elasticsuite-virtual-category/Model/Rule.php:324]: I thought we might don't want to return "null" query anymore that can have a quite unexpected behavior ?
The opposite of forcing an empty result for empty disabled category should be to treat them as normal categories no ?

File: src/module-elasticsuite-virtual-category/Model/Rule.php (hunk 3 above)
Comment (PierreGauthier) [reply src/module-elasticsuite-virtual-category/Model/Rule.php:199]: But if we add it to the cache key, it will invalidate all the category cache if the user update this config as none of the cache key will match anymore ?

**Comment by rbayet** [comment]
We need to discuss this some more.
